### PR TITLE
simplify op name determination into a single forward pass

### PR DESCRIPTION
**Pull request by dagitses**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64261

Note that this does not preserve byte-for-byte compatibility with
existing names.

Test Plan:
* Rely on CI to catch gross errors.
* Merge after release cut to catch subtle issues.

Differential Revision: [D30700647](https://our.internmc.facebook.com/intern/diff/D30700647)

### CODE REVIEW

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30700647).

File: tools/autograd/load_derivatives.py [hunk 1]
```diff
@@ -43,32 +43,15 @@ def load_derivatives(derivatives_yaml_path: str, native_yaml_path: str) -> Seque
             assert str(function.func) not in functions_by_schema
             functions_by_schema[str(function.func)] = function
 
+        # Keep track of how many of which ops we've seen so we can
+        # disambiguate them with a numeric suffix.
+        op_prefixes = Counter[str]()
```
Comment (albanD) [inline tools/autograd/load_derivatives.py:48]: Shouldn't this be `op_postfixes` ?

**Review by albanD — APPROVED** [review]
LGTM except the naming of prefix vs postfix

File: tools/autograd/load_derivatives.py (hunk 1 above)
Comment (dagitses) [reply tools/autograd/load_derivatives.py:48]: It's kinda both. The keys are the prefixes and the values are the suffixes. Thoughts for a better name?

Comment (albanD) [reply tools/autograd/load_derivatives.py:48]: prefix_to_suffix_map ?

Comment (dagitses) [reply tools/autograd/load_derivatives.py:48]: how about `op_counter`? i think that's even clearer.

**Pull request closed without merging**